includes/Context.php

```diff
@@ -357,14 +358,38 @@ public function get_amp_mode() {
 			return false;
 		}
 
-		$exposes_support_mode = method_exists( 'AMP_Theme_Support', 'get_support_mode' )
-			&& defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
-			&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
-			&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
+		if ( defined( 'AMP__VERSION' ) ) {
+			$amp_plugin_version       = AMP__VERSION;
+			$amp_exposes_support_mode = defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
+				&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
+				&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
```

Comment:
This bit needs to remain outside of the clause, otherwise it could be undefined further below.